addons/bus/static/tests/bus_tests.js

```diff
@@ -0,0 +1,165 @@
+});
+
+
+QUnit.module('Bus', {
+    beforeEach: function () {
+        LocalStorageServiceMock = AbstractStorageService.extend({storage: new RamStorage()});
+    },
+}, function () {
+    QUnit.test('notifications received from the longpolling channel', function (assert) {
+        assert.expect(6);
+
+        var pollDeferred = $.Deferred();
+
+        var parent = new Widget();
+        testUtils.addMockEnvironment(parent, {
+            data: {},
+            services: {
+                bus_service: BusService,
+                local_storage: LocalStorageServiceMock,
+            },
+            mockRPC: function (route, args) {
+                if (route === '/longpolling/poll') {
+                    assert.step([route, args.channels.join(',')]);
+
+                    pollDeferred = $.Deferred();
+                    pollDeferred.abort = (function () {
+                        this.reject({message: "XmlHttpRequestError abort"}, $.Event());
+                    }).bind(pollDeferred);
+                    return pollDeferred;
+                }
+                return this._super.apply(this, arguments);
+            }
+        });
+
+        var widget = new Widget(parent);
+        widget.appendTo($('#qunit-fixture'));
+
+        widget.call('bus_service', 'onNotification', this, function (notifications) {
+            assert.step(['notification', notifications]);
+        });
+        widget.call('bus_service', 'addChannel', 'lambda');
+
+        pollDeferred.resolve([{
+            id: 1,
+            channel: 'lambda',
+            message: 'beta',
+        }]);
+        pollDeferred.resolve([{
+            id: 2,
+            channel: 'lambda',
+            message: 'epsilon',
+        }]);
+
+        assert.verifySteps([
+            ["/longpolling/poll", "lambda"],
+            ["notification", [["lambda","beta"]]],
+            ["/longpolling/poll", "lambda"],
+            ["notification", [["lambda","epsilon"]]],
+            ["/longpolling/poll", "lambda"]
+        ]);
+
+        parent.destroy();
+    });
+
+    QUnit.test('cross tab bus share message from a channel', function (assert) {
+        var done = assert.async();
+        assert.expect(5);
+
+        // master
+
+        var pollDeferredMaster = $.Deferred();
+
+        var parentMaster = new Widget();
+        testUtils.addMockEnvironment(parentMaster, {
+            data: {},
+            services: {
+                bus_service: BusService,
+                local_storage: LocalStorageServiceMock,
+            },
+            mockRPC: function (route, args) {
+                if (route === '/longpolling/poll') {
+                    assert.step(['master', route, args.channels.join(',')]);
+
+                    pollDeferredMaster = $.Deferred();
+                    pollDeferredMaster.abort = (function () {
+                        this.reject({message: "XmlHttpRequestError abort"}, $.Event());
+                    }).bind(pollDeferredMaster);
```

Comment:
You don't need to bind to `pollDeferredMaster`, because `this` in this function is correctly linked to `pollDeferredMaster`

Reply:
After testing, it seems to produce an error on phantomjs on the runbot

Reply:
You have to do this because mockRPC returns a `$.Deferred`, whereas the abort method is passed to a promise, and a promise does not have reject or resolve methods.

Here is where you turn the deferred into a promise: https://github.com/odoo-dev/odoo/blob/master-bus-chm/addons/web/static/tests/helpers/mock_server.js#L115

I think you may avoid doing the `bind` in the test by binding the abort method of the promise from the deferred.